metricbeat/module/windows/wmi/utils.go

```diff
@@ -48,9 +48,54 @@ func ConvertSint64(v string) (interface{}, error) {
 	return strconv.ParseInt(v, 10, 64)
 }
 
+const WMI_DATETIME_LAYOUT string = "20060102150405.999999"
+const TIMEZONE_LAYOUT string = "-07:00"
+
+// The CIMDateFormat is defined as "yyyymmddHHMMSS.mmmmmmsUUU".
+// Example: "20231224093045.123456+000"
+// More information: https://learn.microsoft.com/en-us/windows/win32/wmisdk/cim-datetime
+//
+// The "yyyyMMddHHmmSS.mmmmmm" part can be parsed using time.Parse, but Go's time package does not support parsing the "sUUU"
+// part (the sign and minute offset from UTC).
+//
+// Here, "s" represents the sign (+ or -), and "UUU" represents the UTC offset in minutes.
+//
+// The approach for handling this is:
+// 1. Extract the sign ('+' or '-') from the string.
+// 2. Normalize the offset from minutes to the standard `hh:mm` format.
+// 3. Concatenate the "yyyyMMddHHmmSS.mmmmmm" part with the normalized offset.
+// 4. Parse the combined string using time.Parse to return a time.Date object.
 func ConvertDatetime(v string) (interface{}, error) {
-	layout := "20060102150405.999999-0700"
-	return time.Parse(layout, v+"0")
+	if len(v) != 25 {
+		return nil, fmt.Errorf("datetime is invalid: the datetime is expected to be exactly 25 characters long, got: %s", v)
+	}
+
+	// Extract the sign (either '+' or '-')
```

Comment:
@herrBez : WE have changed the date coversion logic here ?

Reply:
Yes, because it turned out I did not get the conversion right (the timezone offset at least). Before the change, we did only support UTC timezone and now we support any timezone